tests/test_temporal.py

```diff
@@ -1,8 +1,5 @@
-from datetime import datetime
```

Comment:
I removed a lot of the unit tests for private methods. I learned that testing private methods is unnecessary and introduces coupling to the implementation details, which breaks encapsulation. It makes refactoring significantly more tedious because the tests for the private methods need to also be updated.

Instead, tests should revolve around public methods and their behaviors/outputs.

More info on why testing private methods is considered "bad" practice":
https://softwareengineering.stackexchange.com/a/380296
https://stackoverflow.com/questions/105007/should-i-test-private-methods-or-only-public-ones